provider/terraform/website/docs/version_2_upgrade.html.markdown

```diff
@@ -0,0 +1,176 @@
+
+~> **NOTE:** This upgrade guide is a work in progress and will not be completed until the release of version 2.0.0 of the provider later this year. Many of the topics discussed, except for the actual provider upgrade, can be performed using the most recent 1.X version of the provider.
+
+Version 2.0.0 of the Google provider for Terraform is a major release and includes some changes that you will need to consider when upgrading. This guide is intended to help with that process and focuses only on changes from version 1.X to version 2.0.0.
+
+Most of the changes outlined in this guide have been previously marked as deprecated in the Terraform plan/apply output throughout previous provider releases. These changes, such as deprecation notices, can always be found in the [Terraform Google Provider CHANGELOG](https://github.com/terraform-providers/terraform-provider-google/blob/master/CHANGELOG.md).
+
+Upgrade topics:
+
+<!-- TOC depthFrom:2 depthTo:2 -->
+
+- [Provider Version Configuration](#provider-version-configuration)
+- [Beta provider](#beta-provider)
+- [Open in Cloud Shell](#open-in-cloud-shell)
+- [Resource: google_bigtable_instance](#resource-google_bigtable_instance)
+- [Resource: google_compute_instance](#resource-google_compute_instance)
+- [Resource: google_compute_instance_from_template](#resource-google_compute_instance_from_template)
+- [Resource: google_compute_project_metadata](#resource-google_compute_project_metadata)
+- [Resource: google_compute_url_map](#resource-google_compute_url_map)
+- [Resource: google_project_iam_policy](#resource-google_project_iam_policy)
+- [Resource: google_sql_database_instance](#resource-google_sql_database_instance)
+- [Resource: google_storage_default_object_acl](#resource-google_storage_default_object_acl)
+- [Resource: google_storage_object_acl](#resource-google_storage_object_acl)
+
+<!-- /TOC -->
+
+## Provider Version Configuration
+
+!> **WARNING:** This topic is placeholder documentation until version 2.0.0 is released later this year.
+
+-> Before upgrading to version 2.0.0, it is recommended to upgrade to the most recent 1.X version of the provider and ensure that your environment successfully runs [`terraform plan`](https://www.terraform.io/docs/commands/plan.html) without unexpected changes or deprecation notices.
+
+It is recommended to use [version constraints when configuring Terraform providers](https://www.terraform.io/docs/configuration/providers.html#provider-versions). If you are following that recommendation, update the version constraints in your Terraform configuration and run [`terraform init`](https://www.terraform.io/docs/commands/init.html) to download the new version.
+
+For example, given this previous configuration:
+
+```hcl
+provider "google" {
+  # ... other configuration ...
+
+  version = "~> 1.19.0"
+}
+```
+
+An updated configuration:
+
+```hcl
+provider "google" {
+  # ... other configuration ...
+
+  version = "~> 2.0.0"
+}
+```
+
+## Beta provider
```

Comment:
I wouldn't hate taking some of the content from Provider Versions and duplicating it here. Obviously the content falling out of sync is a concern, but I think that's acceptable given that this guide is clearly made at a specific point in time.

We can showcase an upgrade of `google_compute_instance` from `google` to `google-beta`, and discuss why the change was made briefly (import, explicit opt-in to beta lack of slas/deprecation, etc) 

My reasoning is mostly aesthetic though, so I'm only really weakly for this - my guess is that the switch to `google-beta` is the change with the most impact, and I'd prefer it to have a heftier section in that case.